src/logic/playercommand.cc

```diff
@@ -2206,24 +2218,28 @@ void CmdDiplomacy::execute(Game& game) {
 			// Note that in the response the numbers of the two players
 			// are swapped compared to the original message.
 			if (it->action == original_action && it->sender == other_player_ &&
-			    it->other == sender()) {
+			    it->other == cmd_sender) {
 				const bool accept =
 				   action_ == DiplomacyAction::kAcceptJoin || action_ == DiplomacyAction::kAcceptInvite;
 				broadcast_message(
 				   accept ? _("Team Change Accepted") : _("Team Change Rejected"),
-				   format(accept ? original_action == DiplomacyAction::kJoin ?
-                               _("%1$s has accepted %2$s into their team.") :
-                               _("%1$s has accepted the invitation to join the team of %2$s.") :
+				   format(accept ?
+                         original_action == DiplomacyAction::kJoin ?
+                         _("%1$s has accepted %2$s into their team.") :
+                         _("%1$s has accepted the invitation to join the team of %2$s.") :
 				          original_action == DiplomacyAction::kJoin ?
-                               _("%1$s has denied %2$s membership in their team.") :
-                               _("%1$s has rejected the invitation to join the team of %2$s."),
-				          sending_player.get_name(), game.get_safe_player(other_player_)->get_name()));
+                         retract ? _("%1$s has retracted the request to join the team of %2$s.") :
+                                   _("%1$s has denied %2$s membership in their team.") :
+				          retract ? _("%1$s has retracted the invitation to %2$s to join their team.") :
+                                _("%1$s has rejected the invitation to join the team of %2$s."),
+				          sending_player.get_name(),
+				          game.get_safe_player(retract ? cmd_sender : other_player_)->get_name()));
```

Comment:
Could you please separate these nested conditional operators into helper variables to make this human readable?